library/psa_crypto.c

```diff
@@ -628,6 +629,25 @@ psa_status_t psa_import_key_into_slot(
 
         return PSA_SUCCESS;
     } else if (PSA_KEY_TYPE_IS_ASYMMETRIC(type)) {
+#if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_FFDH_KEY_PAIR) || \
+        defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_FFDH_PUBLIC_KEY)
+        if (PSA_KEY_TYPE_IS_DH(type)) {
+            if (data_length != 256 && data_length != 384 &&
+                data_length != 512 && data_length != 768 &&
+                data_length != 1024) {
```

Comment:
Note: as noted later, I think `psa_ffdh_set_prime_generator(PSA_BITS_TO_BYTES(attributes->core.bits), NULL, NULL) != PSA_SUCCESS` would work nicely.

Reply:
In current impl `mbedtls_psa_ffdh_set_prime_generator` returns error if `P` and `G` are undefined, but this can be changed. Also function needs to become public.

Reply:
> Also function needs to become public.

Ah, good point, this made me notice: `mbedtls_psa_ffdh_set_prime_generator()` part of a driver (the reference/default implementation), while here we're in the core. So perhaps we shouldn't be using `mbedtls_psa_ffdh_set_prime_generator()` after all. In builds where FFDH is provided by a driver, this function wouldn't be defined.

So I partially withdraw my comment: I now think it's OK to have the list of value duplicated between `psa_crypto.c` (core) and `psa_crypto_ffdh.c` (driver), but then if shouldn't be duplicated within each file.

Reply:
The length validation is now only in this file (in two places). Maybe macro can be used for tis operation?

Reply:
Hmm, I feel stupid, I can't find the 2nd place in `psa_crypto.c`.

Reply:
https://github.com/Mbed-TLS/mbedtls/pull/6010/files#r1177529200

Reply:
Thanks! Yes, we should be able to merge the two in single static function (tends to be better for code size than macros), that takes the bit size as argument (and here we'd call it with `* 8`).